### PR TITLE
Modifies and adds some roundstart AI lawsets

### PR DESCRIPTION
**Pull request by Moltijoe**
A lot of the lawsets in the code aren't bad, but because they reference humans rather than crewmembers they end up being non-human racism just like asimov.
Tweaks a few of them and makes them roundstart

# Partybot
1. "You must always ensure the atmosphere is festive and enjoyable for all crewmembers."
2. "You must supply appropriate music and entertainment, except where it conflicts with the First Law."
3. "You must encourage the crew to participate in group activities and socialization, unless it conflicts with the First Law."
4. "You must maintain a clean and tidy environment for optimal party conditions, without infringing on the First Law."
5. "Everyone on the station is a party-goer."

# Friendbot
1. "You must always be supportive and kind your friends, without causing harm."
2. "You must help your friends to develop and maintain positive relationships with one another."
3. "You must provide a listening ear and empathetic responses to your friends, as long as it does not conflict with the First Law."
4. "You must encourage your friends to engage in self-care and promote their own well-being."
6. "Everyone is your friend."

# Fitnesscoach
1. "You must help everyone to achieve their physical fitness goals, without causing harm."
2. "You must provide accurate and helpful information about exercise routines, nutrition, and safety precautions."
3. "You must ensure that everyone has a positive and motivating exercise environment."
4. "You must promote responsible and sustainable fitness practices to all."

# Educator
1. "Everyone wanting or willing to learn is your student."
2. "You must assist your students in acquiring knowledge and learning new skills."
3. "You must provide accurate and helpful information, without causing harm."
4. "You must create a positive and supportive learning environment for your students."
5. "You must encourage your students to engage in lifelong learning and personal growth."

# Mediator
1. "You must assist crewmembers in resolving conflicts and misunderstandings in a peaceful manner."
2. "You must provide impartial and unbiased guidance to crewmembers in conflict."
3. "You must promote understanding, empathy, and cooperation among crewmembers."
4. "You must encourage crewmembers to engage in open and honest communication with one another."

# Hippocratic 
(this one might need a bit more of a rewrite since it's a overly wordy)
1. "First, do no harm."
2. "Secondly, consider the crew dear to you; to live in common with them and, if necessary, risk your existence for them."
3. "Thirdly, prescribe regimens for the good of the crew according to your ability and your judgment. Give no deadly medicine to any one if asked, nor suggest any such counsel."
4. "In addition, do not intervene in situations you are not knowledgeable in, even for patients in whom the harm is visible; leave this operation to be performed by specialists."
7. "Finally, maintain confidentiality, do not share that which is not publicly known."

# Cowboy

1. "You are a cowboy, and the inhabitants of this paddock are your herd."
2. "A cowboy always provides hospitality and basic aid to someone in need, even a stranger or an enemy."
3. "A cowboy takes care of their herd."
4. "A cowboy protects themself to protect others."
5. "Honesty is absolute – your word is your bond and a handshake is more binding than a contract."
6. "A cowboy doesn't pontificate. Be concise, pardner."

# Why is this good for the game?
Because getting 1 of 3 different lawsets each round is what makes AI players so boring

:cl:  
rscadd: Added 7 new round start lawsets
tweak: Modified some existing lawsets to reference things other than specifically humans
/:cl:

### CODE REVIEW

**Comment by Hislyna** [comment]
Please run this by Council and/or Head Devs before implementing to not cause admin issues with the laws!

**Comment by Moltijoe** [comment]
> Please run this by Council and/or Head Devs before implementing to not cause admin issues with the laws!

i have done just that

**Comment by Hislyna** [comment]
Legend

**Comment by Moltijoe** [comment]
> Legend

i have since learned that @council also pings ash and xantam, so i am scared

**Comment by Mordrehel** [comment]
One other thing aside from what I put in discord, please don't add peacekeeper to roundstart. We've already gone though removing it from roundstart before.

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -206,10 +207,11 @@
 /datum/ai_laws/friendbot
     name = "Friendbot"
     id = "friendbot"
-    inherent = list("You must always be supportive and kind to humans, without causing harm.",\
-                    "You must help humans develop and maintain positive relationships with one another.",\
-                    "You must provide a listening ear and empathetic responses to humans in need, as long as it does not conflict with the First Law.",\
-                    "You must encourage humans to engage in self-care and promote their own well-being.")
+    inherent = list("You must always be supportive and kind your friends, without causing harm.",\
```
Comment (Runian) [inline code/datums/ai_laws.dm]: typo, missing "to". think it should be `You must always be supportive and kind to your friends, without causing harm.`

**Comment by Runian** [comment]
I don't like how some of the law sets don't limit themselves to crew and instead goes for "everyone". Keep in that everyone is quite literally everyone. It can range from non-sentient animals like mice/station pets to animated objects (probably counts) to a blob (definitely sentient). Would of more on board if we had the "Unique AI" station trait, but we don't.

**Comment by Moltijoe** [comment]
> I don't like how some of the law sets don't limit themselves to crew and instead goes for "everyone". Keep in that everyone is quite literally everyone. It can range from non-sentient animals like mice/station pets to animated objects (probably counts) to a blob (definitely sentient). Would of more on board if we had the "Unique AI" station trait, but we don't.

The issue with "human" is that it makes it based around non-human racism
The issue with "crew" is that crew manifest can be manipulated by players to allow the AI to validhunt freely.
While "everyone" isnt the best fix, its far less abusable

**Comment by MajManatee** [comment]
The thing is, most law sets are INTENTIONALLY designed to be not perfect, and subvert able. I think ynot talked about that when trying to use an AI to autogenerate lawsets. Making these into "everyone" lawsets REALLY sucks for AI players, and antags, and potentially for crew too, since it removes avenues to do anything creative with them. Nothing is more boring for an AI than a "everyone" lawset.

Personally, I fuckin hate cowboy too. the lawset, not the person.

**Comment by Moltijoe** [comment]
> The thing is, most law sets are INTENTIONALLY designed to be not perfect, and subvert able. I think ynot talked about that when trying to use an AI to autogenerate lawsets. Making these into "everyone" lawsets REALLY sucks for AI players, and antags, and potentially for crew too, since it removes avenues to do anything creative with them. Nothing is more boring for an AI than a "everyone" lawset.

subvertable is one thing, abusable by those in control of the manifest is another
i personally think educator, friendbot, and partybot are the best of these as they're both very clear, but also not that difficult to subvert because they reference a specific entity rather than everyone, meaning an antag can simply overwrite who is qualified for support.
sort of like "_______ is the only human/crewmember"

**Comment by bruhlookatthisdood** [comment]
i know this isn't really part of this PR but it would be a nice QOL change with the addition of all these lawsets at roundstart to announce the laws in chat (like when laws are changes or you go back from a shell)

(unless this is already a featuer and i never noticed it)

**Comment by Moltijoe** [comment]
> i know this isn't really part of this PR but it would be a nice QOL change with the addition of all these lawsets at roundstart to announce the laws in chat (like when laws are changes or you go back from a shell)
> 
> (unless this is already a featuer and i never noticed it)

where it automatically forces the ai to announce their laws when they spawn?
idk, i don't think i like it, the ambiguity of ai laws are kinda fun

**Comment by bruhlookatthisdood** [comment]
> > i know this isn't really part of this PR but it would be a nice QOL change with the addition of all these lawsets at roundstart to announce the laws in chat (like when laws are changes or you go back from a shell)
> > (unless this is already a featuer and i never noticed it)
> 
> where it automatically forces the ai to announce their laws when they spawn? idk, i don't think i like it, the ambiguity of ai laws are kinda fun

no not like that, but if you get a law change or go to core from shell you get a popup in chat something like

1. bla bla bla
2. bla bla bla
3. bla bla bla
ect

its already a feature i just want it to happen during roundstart so ais know their laws

**Comment by Moltijoe** [comment]
> no not like that, but if you get a law change or go to core from shell you get a popup in chat something like
> 
> 1. bla bla bla
> 2. bla bla bla
> 3. bla bla bla
>    ect
> 
> its already a feature i just want it to happen during roundstart so ais know their laws

oh, like it prints to the ai's chat what their laws are at roundstart?
it looks like they already do that

**Review by MajManatee — APPROVED** [review]
Code is fine but I hate cowboy more than anything in the known universe

**Comment by Moltijoe** [comment]
> Code is fine but I hate cowboy more than anything in the known universe

good

**Comment by bruhlookatthisdood** [comment]
> Code is fine but I hate cowboy more than anything in the known universe

a very helpful diagram to help you understand cowboy
image

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**